docs/source/tutorial/glance.rst

```diff
@@ -0,0 +1,405 @@
+* Define by Run -- neural networks definitions are defined on-the-fly at run time, allowing for dynamic network changes
+* NumPy based syntax for working with arrays, thanks to CuPy implementation
+* Fully customizable -- since Chainer is pure python, all classes and methods can be adapted to allow for the latest cutting edge or specialized approaches
+* Broad and deep support -- Chainer is actively used for most of the current approaches for neural nets (CNN, RNN, RL, etc.), aggressively adds new approaches as they're developed, and provides support for many kinds of hardware as well as parallelization for multiple GPUs
+
+
+Mushrooms -- tasty or deathly?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's take a look at a basic program of Chainer to see how it works. For a dataset, we'll work with `Kaggle's edible vs. poisonous mushroom dataset <https://www.kaggle.com/uciml/mushroom-classification>`_, which has over 8,000 examples of mushrooms, labelled by 22 categories including odor, cap color, habitat, etc.
+
+How will Chainer learn which mushrooms are edible and which mushrooms will kill you? Let's see!
+
+Full Code
+---------------
+
+Here's the whole picture of the code:
+
+.. testcode::
+    
+   #!/usr/bin/env python
+   
+   import matplotlib
+   matplotlib.use('Agg')
+
+   import chainer
+   import chainer.functions as F
+   import chainer.links as L
+   from chainer import datasets
+   from chainer import training
+   from chainer.training import extensions
+   
+   import numpy as np
+   
+   data_array = np.genfromtxt(
+       'mushrooms.csv', delimiter=',', dtype=str, skip_header=1)
+   for col in range(data_array.shape[1]):
+       data_array[:, col] = np.unique(data_array[:, col], return_inverse=True)[1]
+   
+   X = data_array[:, 1:].astype(np.float32)
+   Y = data_array[:, 0].astype(np.int32)[:, None]
+   train, test = datasets.split_dataset_random(
+       datasets.TupleDataset(X, Y), int(data_array.shape[0] * .7))
+   
+   train_iter = chainer.iterators.SerialIterator(train, 100)
+   test_iter = chainer.iterators.SerialIterator(
+       test, 100, repeat=False, shuffle=False)
+   
+   
+   # Network definition
+   class MLP(chainer.Chain):
+       def __init__(self, n_units, n_out):
+           super(MLP, self).__init__()
+           with self.init_scope():
+               # the input size to each layer inferred from the layer before
+               self.l1 = L.Linear(n_units)  # n_in -> n_units
+               self.l2 = L.Linear(n_units)  # n_units -> n_units
+               self.l3 = L.Linear(n_out)  # n_units -> n_out
+   
+       def __call__(self, x):
+           h1 = F.relu(self.l1(x))
+           h2 = F.relu(self.l2(h1))
+           return self.l3(h2)
+   
+   
+   model = L.Classifier(
+       MLP(44, 1), lossfun=F.sigmoid_cross_entropy, accfun=F.binary_accuracy)
+   
+   # Setup an optimizer
+   optimizer = chainer.optimizers.SGD()
+   optimizer.setup(model)
```

Comment:
How about removing this line and use method chain? See https://github.com/chainer/chainer/pull/4141